test/widgets/app_test.dart

```diff
@@ -51,4 +54,113 @@ void main() {
+        await testBinding.globalStore
+          .insertAccount(account.toCompanion(false));
+      }
+
+      await tester.pumpWidget(
+        const MaterialApp(
+          localizationsDelegates: ZulipLocalizations.localizationsDelegates,
+          supportedLocales: ZulipLocalizations.supportedLocales,
+          home: GlobalStoreWidget(
+            child: ChooseAccountPage())));
+
+      // global store gets loaded
+      await tester.pumpAndSettle();
+    }
+
+    List<Account> generateAccounts(int count) {
+      return List.generate(count, (i) => eg.account(
+        id: i,
+        user: eg.user(fullName: 'User $i', email: 'user$i@example'),
+        apiKey: 'user${i}apikey',
+      ));
+    }
+
+    Finder findAccount(Account account) => find.text(account.email).hitTestable();
+
+    Finder findButton<T extends ButtonStyleButton>({required String withText}) {
+      return find
+        .descendant(of: find.bySubtype<T>(), matching: find.text(withText))
+        .hitTestable();
+    }
+
+    void checkAccountShown(Account account, {required bool expected}) {
+      check(findAccount(account).evaluate()).length.equals(expected ? 1 : 0);
+    }
+
+    void checkButtonShown<T extends ButtonStyleButton>({
+      required String withText,
+      required bool expected,
+    }) {
+      check(findButton<T>(withText: withText).evaluate())
+        .length.equals(expected ? 1 : 0);
+    }
+
+    testWidgets('accounts list is scrollable when more than a screenful',
+      (tester) async {
+      final accounts = generateAccounts(15);
+      await setupChooseAccountPage(tester, accounts: accounts);
+
+      // Accounts list is more than a screenful
+      //  * First account is shown
+      //  * Last account is out of view
+      checkAccountShown(accounts.first, expected: true);
+      checkAccountShown(accounts.last, expected: false);
+
+      // Button to add an account is visible
+      // and not moved offscreen by the long list of accounts
+      checkButtonShown(withText: 'Add an account', expected: true);
+
+      // Accounts list is scrollable to the bottom
+      await tester.scrollUntilVisible(findAccount(accounts.last), 50);
+      checkAccountShown(accounts.last, expected: true);
+    });
+
+    testWidgets('with just one account, the layout is centered',
+      (tester) async {
+      final account = eg.selfAccount;
+      await setupChooseAccountPage(tester, accounts: [account]);
+
+      const buttonText = 'Add an account';
+      checkAccountShown(account, expected: true);
+      checkButtonShown(withText: buttonText, expected: true);
+
+      final screenHeight =
+        (tester.view.physicalSize / tester.view.devicePixelRatio).height;
+      final thirdOfScreenHeight = screenHeight / 3;
+
+      final accountRect = tester.getRect(findAccount(account));
+      check(accountRect.top).isGreaterThan(thirdOfScreenHeight);
+      check(accountRect.bottom).isLessThan(screenHeight - thirdOfScreenHeight);
```

Comment:
Let's write this in more thoroughly `package:checks` style:
```suggestion
      check(tester.getRect(findAccount(account)))
        ..top.isGreaterThan(thirdOfScreenHeight)
        ..bottom.isLessThan(screenHeight - thirdOfScreenHeight);
```

You'll need to write those accessors `top` and `bottom` , since we haven't yet. They'll go in `test/flutter_checks.dart`; see existing examples there.